src/data/commands.js

```diff
@@ -137,6 +152,39 @@ export function completeQuest(dataName) {
             FONTS.ITEM_STATS
         );
     } else {
-        chatScene.writeText(`The level '${dataName}' does not exist.`);
+        chatScene.writeText(`The level '${levelName}' does not exist.`);
+    }
+}
+
+export function unlockSong(songName) {
+    // Song unlocks are currently tied to quest completion, so if we want
+    // to be able to unlock songs independently, we will need to decouple
+    // them in the future
+
+    const chatScene = characterData.getScene(CONSTANTS.SCENES.CHAT);
+    const dashScene = characterData.getScene(CONSTANTS.SCENES.DASHBOARD);
+
+    const constName = songName.toLowerCase();
+    if (constName == "all") {
+        // unlock all songs
+        dashScene.music.songTexts.forEach((textObj) => {
+            characterData.setQuestCompleted(textObj.prereq);
+            // Let the music scene handle the rest
+        });
+        return;
+    }
+
+    const songTextObj = dashScene.music.songTexts.find((textObj) => {
```

Comment:
Personal style preference for short lambdas:
```suggestion
    const songTextObj = dashScene.music.songTexts.find((textObj) => textObj.text == constName);
```

Reply:
https://github.com/EthanVieira/rs-clicker/pull/75/commits/820e93bc6455d15beceb0ea3d55917c523df4df2